lightning/src/routing/scoring.rs

```diff
@@ -637,6 +640,22 @@ pub struct ProbabilisticScoringFeeParameters {
 	///
 	/// Default value: false
 	pub linear_success_probability: bool,
+
+	/// In order to ensure we have knowledge for as many paths as possible, when probing it makes
+	/// sense to bias away from channels for which we have very recent data.
+	///
+	/// This value is a penalty that is applied based on the last time that we updated the bounds
+	/// on the available liquidity in a channel. The specified value is the maximum penalty that
+	/// will be applied.
+	///
+	/// It obviously does not make sense to assign a non-0 value here unless you are using the
+	/// pathfinding result for background probing.
```

Comment:
This way of probing where you set this exploration/exploitation trade-off is just one way to do it, isn't it?

From https://lightningdevkit.org/probing:

>Recall that earlier, we discussed the concept of using multiple scorers. One scorer that we actually save results to and another un-updated scorer that can be used to identify routes with a "beginner's mind" - so to speak. If we'd like to leverage this approach, we can pass an un-updated scorer here and then update a different scorer with the results of the probe.

This is a different approach where no scores are used for probe route selection, which is different again from what is in the PR description (using a single scorer).

What's your current thinking on what's best?

Reply:
I can follow your thoughts there, but I am quite uncomfortable with the level of gut feeling involved in these decisions. This is probably something that keeps coming back. In https://github.com/lightningdevkit/rust-lightning/issues/2709 we were also guessing a bit at the best approach, and there are probably more areas where it's not very clear what the best approach or parameters are.

It does make me think more about the idea of setting up a test framework that involves a virtual network of nodes.

Reply:
Maybe that is true. With a sim network though, I think the discussion will move to a higher level. We'd discuss whether the network is realistic enough, and parameters / probe methods / pathfinding algorithms will follow from that.

To me, A/B testing sounds even more prone to getting non-representative results and is probably too slow to scan parameter spaces.